src/shogun/lib/SGMatrix.cpp

```diff
@@ -48,6 +48,7 @@ SGMatrix<T>::SGMatrix(index_t nrows, index_t ncols, bool ref_counting)
 	: SGReferencedData(ref_counting), num_rows(nrows), num_cols(ncols)
 {
 	matrix=SG_MALLOC(T, ((int64_t) nrows)*ncols);
+	zero();
```

Comment:
Yes indeed that would be slower (when we init the matrix later and
besides that one could use SG_CMALLOC if we really wanted this)

On Wed, 2015-03-25 at 03:44 -0700, Heiko Strathmann wrote:
> > @@ -48,6 +48,7 @@ SGMatrix<T>::SGMatrix(index_t nrows, index_t ncols, bool ref_counting)
> >  	: SGReferencedData(ref_counting), num_rows(nrows), num_cols(ncols)
> >  {
> >  	matrix=SG_MALLOC(T, ((int64_t) nrows)*ncols);
> > +	zero();
> 
> why that?
> We dont want that I think
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/shogun-toolbox/shogun/pull/2781/files#r27109755